docs/section21/Rule21-9.md

```diff
@@ -0,0 +1,38 @@
+
+# Boiler - Rule 21-9  
+
+**Rule ID:** 21-9  
+**Rule Description:** When baseline building includes boilers, Hot Water Pump Power = 19W/gpm.  
+**Rule Assertion:** B-RMR  
+**Appendix G Section:** Section 21 Boiler  
+**Appendix G Section Reference:** Section G3.1.3.5 Building System-Specific Modeling Requirements for the Baseline model  
+
+**Applicability:** All required data elements exist for B_RMR  
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-1, 5, 7, 11, or 12
+2. B-RMR is not modeled with purchased heating.
+
+**Manual Check:** None  
+**Evaluation Context:** Building  
+**Data Lookup:** None  
+**Function Call:** None  
+
+**Applicability Checks:**  
+
+1. B-RMR is modeled with at least one air-side system that is Type-1, 5, 7, 11, or 12: `if PLACEHOLDER_AIRSIDE_SYSTEM_RULE-X-XX:`
+2. B-RMR is not modeled with purchased heating: `if NOT Rule-21-1:`
+
+## Rule Logic:  
+
+- For each fluid loop in B_RMR: `for fluid_loop_b in B_RMR.ASHRAE229.fluid_loops:`
+
+  - Check if fluid loop is heating type: `if fluid_loop_b.type == "HEATING":`
+
+    **Rule Assertion:**
+
+    - Case 1: For each heating hot water loop that is served by boiler(s), if total pump power per flow rate is equal to 19W/gpm: `if fluid_loop_b.pump_power_per_flow_rate == 19: PASS`
```

Comment:
@xingcx a fluid loop can have multiple child loops. we need to go and sum the pump power of all child loops and verify that the total pump power off all child loops is =19 W/CFM

Reply:
Updated.

Reply:
Added Rule 21-11 (Baseline shall have only one HHW loop) and 21-18 (Baseline shall have only one primary HHW loop and no secondary loop) as applicability check. So now we do not need to check child loop pump power for this rule.

Reply:
The applicability check is updated. Right now the logic only checks pump power on heating loops that are connected to boiler(s).